### ISSUE TITLE
systemProbe.enabled deprecation breaks upgrades with Flux

### ISSUE THREAD

**Issue by Niksko** [DataDog/helm-charts#174]
**Describe what happened:**
I tried to upgrade from 2.8.5 to 2.9.0, and the datadog Helm chart wouldn't upgrade because of the deprecation of `systemProbe.enabled`

**Describe what you expected:**
A major version bump, because this update is breaking.

**Steps to reproduce the issue:**
1. Create a `values.yaml` file with `systemProbe.enabled` set to `true`
2. Install version 2.8.5 of the Datadog Helm chart
3. Upgrade to 2.9.0 of the Datadog Helm chart

**Additional environment details (Operating System, Cloud provider, etc):**
EKS 1.18, Flux v2

I believe this is because the deprecation warning includes the `fail` directive. I think including the deprecation warning is fine, but including the `fail` is not.

**Comment by clamoriniere** [comment]
Hi @Niksko 

Sorry for the inconvenience.

The deprecation `systemProbe.enabled` notice was a warning since few versions already. Keeping this parameter was a source of unknown state in the systemProbe process, that is why we move it to a fail state when `systemProbe.enabled:true`

We believe this change is a step in the right direction. It is more clear for the user which feature is enabled with the 3 distinct options that can be enabled separately.

If previously, you only set `systemProbe.enabled: true` you can replace it by using` networkMonitoring.enabled:true`.

Let me know if you need more support.

**Comment by Niksko** [comment]
Thanks for the context. My issue isn't so much the deprecation. I understand why it was done, and we've successfully adopted the required changes.

My issue is more one if semver and stability. Given there has been no update to version 3 of the chart, I would have expected no breaking changes. This was a breaking change for us.

A deprecation is fine, but I guess I don't really understand why this absolutely had to be removed. It seems like for users like us that hadn't noticed the deprecation warning (this includes anyone using Flux, as Flux doesn't seem to surface notes.txt info unless it includes a `fail` like this one), still having this parameter wasn't having any adverse effects, and had been worked around.

Basically, we would have preferred you left this as deprecated until you released version 3, instead of including the failure in the notes.txt

**Comment by clamoriniere** [comment]
They,

We might have increased the major digit, but because this settings wasn't enabled by default, and it was to solve a "undefine" state with recent agent version, we choose to avoid customer behind it by this bug.
We will be more careful on the impact that this kind of change might have on users using Flux or any other automatic deployment solution.

Thanks for your comprehension 🙇‍♂️
